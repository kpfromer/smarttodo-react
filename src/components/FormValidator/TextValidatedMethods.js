import validateIsEmail from "validator/lib/isEmail";

export const required = isRequired => ({
  hint: name => isRequired ? `${name} is required` : '',
  validate: value => isRequired ? !!value : true
});

export const minLength = minValue => ({
  hint: name => `${name} must be ${minValue + 1} characters or more`,
  validate: value => value.length > minValue
});

export const maxLength = maxValue => ({
  hint: name => `${name} must be ${maxValue - 1} characters or less`,
  validate: value => value.length < maxValue
});

export const email = isEmail => ({
  hint: () => isEmail ? `Must be a valid email` : '',
  validate: value => isEmail ? validateIsEmail(value) : true
});