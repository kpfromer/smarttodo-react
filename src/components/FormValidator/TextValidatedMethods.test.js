import * as validatedMethods from "./TextValidatedMethods";

describe('TextValidatedMethods', () => {
  let name = 'hintName';
  [
    {
      name: 'required',
      value: true,
      expected: [
        ['notempty', true],
        ['', false]
      ]
    },
    {
      name: 'minLength',
      value: 4,
      expected: [
        ['four', false],
        ['five5', true]
      ]
    },
    {
      name: 'maxLength',
      value: 6,
      expected: [
        ['sixsix', false],
        ['five', true]
      ]
    },
    {
      name: 'email',
      value: true,
      expected: [
        ['valid@email.com', true],
        ['invalid.com', false],
        ['invalid@email', false]
      ]
    }
  ].forEach(
    ({ name: methodName, value, expected }) => {

      describe(methodName, () => {
        let hintFn, validateFn;

        beforeEach(() => {
          ({ hint: hintFn, validate: validateFn } = validatedMethods[methodName](value));
        });

        it('creates a hint with the input name', () => {
          expect(hintFn(name)).toMatchSnapshot();
        });

        describe('validation', () => {
          expected.forEach(value => {
            const [ inputValue, expectedValue ] = value;
            const message = expectedValue ? 'valid' : 'invalid';
            it(`expect ${inputValue} to be ${message}`, () =>
              expect(validateFn(inputValue)).toBe(expectedValue)
            );
          });
        });
      });
    }
  );

  describe('required', () => {

  });

  describe('minLength', () => {

  });

  describe('maxLength', () => {

  });

  describe('email', () => {

  });
});