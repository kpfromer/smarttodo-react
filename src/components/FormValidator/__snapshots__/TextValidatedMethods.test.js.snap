// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextValidatedMethods email creates a hint with the input name 1`] = `"Must be a valid email"`;

exports[`TextValidatedMethods maxLength creates a hint with the input name 1`] = `"hintName must be 5 characters or less"`;

exports[`TextValidatedMethods minLength creates a hint with the input name 1`] = `"hintName must be 5 characters or more"`;

exports[`TextValidatedMethods required creates a hint with the input name 1`] = `"hintName is required"`;
